### PR TITLE
Update service clients instantiation doc comments

### PR DESCRIPTION
**Pull request by drwill-ms**

### CODE REVIEW

File: iothub/service/src/RegistryManager.cs [hunk 1]
```diff
@@ -70,48 +70,30 @@ public class RegistryManager : IDisposable
         private IHttpClientHelper _httpClientHelper;
 
         /// <summary>
-        /// Creates an instance of <see cref="RegistryManager"/>, provided for unit testing purposes only.
```
Comment (drwill-ms) [inline iothub/service/src/RegistryManager.cs:73]: No need to self-ref to the same topic we're already on.

File: iothub/service/src/RegistryManager.cs [hunk 2]
```diff
@@ -70,48 +70,30 @@ public class RegistryManager : IDisposable
...
+        /// Creates an instance of RegistryManager, provided for unit testing purposes only.
         /// Use the CreateFromConnectionString method to create an instance to use the client.
         /// </summary>
         public RegistryManager()
         {
         }
 
-        internal RegistryManager(IotHubConnectionProperties connectionProperties, HttpTransportSettings transportSettings)
```
Comment (drwill-ms) [inline iothub/service/src/RegistryManager.cs:80]: Moved internals below publics.

File: iothub/service/src/RegistryManager.cs (hunk 1 above)
Comment (azabbasi) [reply iothub/service/src/RegistryManager.cs:73]: The only benefit of self-ref would be to get consistency if we ever rename this class. hint hint

**Review by azabbasi — APPROVED** [review]

File: iothub/service/src/RegistryManager.cs (hunk 1 above)
Comment (drwill-ms) [reply iothub/service/src/RegistryManager.cs:73]: Meh, cost of user confusion is worse IMO.

**Review by timtay-microsoft — APPROVED** [review]

**Review by azabbasi — APPROVED** [review]

**Pull request merged**